templates/ui-frameworks/svelte/web-app/ui/src/contexts.ts.hbs

```diff
@@ -1,7 +1,83 @@
-import { type AppClient } from "@holochain/client";
+import { writable, type Writable } from 'svelte/store';
+import { getContext, setContext } from 'svelte';
+import type { AppClient, HolochainError } from '@holochain/client';
+import { AppWebsocket } from '@holochain/client';
+{{#if holo_enabled}}
+import WebSdk from '@holo-host/web-sdk';
+{{/if}}
 
-export const clientContext = "AppClient";
+export const CLIENT_CONTEXT_KEY = Symbol('holochain-client');
 
-export type ClientContext = {
-  getClient: () => Promise<AppClient>;
-};
+interface ClientStore {
+  client: AppClient | undefined;
+  error: HolochainError | undefined;
+  loading: boolean;
+}
+
+{{#if holo_enabled}}
+const IS_HOLO = ['true', '1', 't'].includes(import.meta.env.VITE_APP_IS_HOLO?.toLowerCase());
+{{/if}}
+
+export function createClientStore() {
+  const store = writable<ClientStore>({
+    client: undefined,
+    error: undefined,
+    loading: false
+  });
+
+  const { subscribe, set, update } = store;
+
+  return {
+    subscribe,
+    connect: async () => {
+      update(s => ({ ...s, loading: true }));
+      try {
+        {{#if holo_enabled}}
+        if (IS_HOLO) {
```

Comment:
Do we still need to keep supporting Holo architecture?

Reply:
I am not sure, the most up to date version of the holo web sdk seems to use [`v0.18.0-dev.2`](https://github.com/Holo-Host/web-sdk/blob/develop/package.json#L51C28-L51C40) of the js client, supporting the 0.4 line. Updates to the lib have since stopped though.

I can bring this up as a topic in the weekly meeting.

Reply:
Yes, please do!